internal/build_defs.bzl

```diff
@@ -184,3 +184,93 @@ def tsc_library(**kwargs):
+
+  Returns:
+    A closure_js_library with the rerooted files.
+  """
+  rerooted_prod_files = collect_es6_sources(ctx)
+
+  js_module_roots = depset()
+  for prod_file in rerooted_prod_files:
+    if "node_modules/" in prod_file.dirname:
+      js_module_roots += [prod_file.dirname[:prod_file.dirname.find('node_modules/')]]
+
+  return struct(
+    files = rerooted_prod_files,
+    closure_js_library = struct(
+      srcs = rerooted_prod_files,
+      js = [
+
+      ],
+      js_module_roots = js_module_roots
+  ))
+
+_collect_es6_sources = rule(
+    attrs = {"deps": attr.label_list(mandatory = True)},
+    implementation = _collect_es6_sources_impl,
+)
+
+def closure_ts_binary(name, deps, **kwargs):
+  _collect_es6_sources_label = name + "_collect_es6_sources"
+  _collect_es6_sources(name = _collect_es6_sources_label, deps = deps)
+
+  closure_js_binary(
+    name = name,
+    deps = [":" + _collect_es6_sources_label],
+    **kwargs
+  )
+
+def closure_ng_binary(name, defs = [], **kwargs):
+  rerooted_node_modules_path = "**/%s_collect_es6_sources.es6/node_modules" % name
+
+  rxjs_path = "%s/rxjs/**.js" % rerooted_node_modules_path
+
+  closure_ts_binary(
+    name = name, 
+    defs = defs + [
+        ### jscomp_off flags
+        # TODO(mrmeku): Remove this flag to declrease binary size further. The key to removing
+        # this variable is making STRICT dependency mode work or by having externs for SystemJS, 
+        # NodeJS and HammerJS (since these are references within the transitive closure of @angular)
+        "--jscomp_off=checkVars",
+        # Needed for RXJS. 
+        # TODO(mrmeku): Investigate whether this flag being needed is a tsickle bug.
+        "--jscomp_off=checkTypes",
+        # Needed for RXJS. 
+        # TODO(mrmeku): Investigate whether this flag being needed is a tsickle bug.
+        "--jscomp_off=jsdocMissingType",
+        # Needed for RXJS. 
+        # TODO(mrmeku): Investigate whether this flag being needed is a tsickle bug.
+        "--jscomp_off=misplacedTypeAnnotation",
+
+        ### Externs
+        "--js=node_modules/@angular/core/src/testability/testability.externs.js",
+
+        ### @angular Dependencies 
+        "--js=node_modules/hammerjs/hammer.js",
+        "--js=%s" % rxjs_path,
+
+        ### @angular packages
+        "--package_json_entry_names=es2015",
+        "--js=node_modules/@angular/**/package.json",
+        # Core must be specified individually and not globbed with packages which depend on it 
+        # (ie. @angular/animations).
+        "--js=node_modules/@angular/core/esm2015/core.js",
+        # Glob the other @angular packages without core.
+        "--js=node_modules/@angular/**/esm2015/*.js,node_modules/@angular/core/esm2015/core.js",
+
+        ### All other filegroup, Closure Library, and rerooted Typescript files.
+        "--js={0}/**.js,!{1},!{0}/**.ngsummary.js".format(rerooted_node_modules_path, rxjs_path),
+    ],
+    manually_specify_js = True,
```

Comment:
This line causes errors for me if I try to use `closure_ng_binary`. It looks like this attribute gets forwarded to `closure_js_binary` which doesn't recognize it.